Python/pythonrun.c

```diff
@@ -1199,6 +1199,38 @@ print_exception_notes(struct exception_print_context *ctx, PyObject *notes)
     return -1;
 }
 
+static int
+get_exception_notes(struct exception_print_context *ctx, PyObject *value, PyObject **notes) {
+    PyObject *type = NULL;
+    PyObject *errvalue = NULL;
+    PyObject *tback = NULL;
+    PyObject *note = NULL;
+
+    if (_PyObject_LookupAttr(value, &_Py_ID(__notes__), notes) < 0) {
+        PyErr_Fetch(&type, &errvalue, &tback);
+        note = PyUnicode_FromFormat("Ignored error getting __notes__: %R", errvalue);
+        if (!note) {
+            goto error;
+        }
+        *notes = PyList_New(1);
+        if (!*notes) {
+            goto error;
+        }
+        if (PyList_SetItem(*notes, 0, note) < 0) {
+            Py_DECREF(*notes);
+            goto error;
+        }
+        Py_XDECREF(type);
+        Py_XDECREF(errvalue);
+        Py_XDECREF(tback);
```

Comment:
These are too late - they won't happen if we error out before we reach this.

Reply:
oh yes, of course, sorry for all these mistakes :) I pushed again to address these comments, thank you.